pulp_rpm/app/serializers/repository.py

```diff
@@ -118,8 +124,45 @@ def validate(self, data):
                 raise serializers.ValidationError({field: _(ALLOWED_CHECKSUM_ERROR_MSG)})
 
         validated_data = super().validate(data)
+        if (data.get("gpgcheck") or data.get("repo_gpgcheck")) and data.get("repo_config"):
+            raise serializers.ValidationError(
+                _(
+                    "Cannot use gpg options and 'repo_config' options simultaneously. "
+                    "The 'gpgcheck' and 'repo_gpgcheck' options aew deprecated, please use "
+                    "'repo_config' only."
+                )
+            )
         return validated_data
 
+    def create(self, validated_data):
+        """
+        Save the repo and handle gpg options
+
+        Args:
+            validated_data (dict): A dict of validated data to create the repo
+
+        Returns:
+            repo: the created repo
+        """
+        gpgcheck = validated_data.pop("gpgcheck", None)
```

Comment:
Leave a comment here about what this is doing, i.e. acting as a shim between old and new APIs.